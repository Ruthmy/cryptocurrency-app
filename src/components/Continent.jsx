/* eslint-disable no-unused-vars */
import React from 'react';
import { useSelector } from 'react-redux';
import { NavLink, useParams } from 'react-router-dom';

import '../styles/Continent.css';

const Continent = () => {
  // Get the continents from the global state
  // const continents = useSelector((state) => state.location.location);
  const { continent } = useParams();
  console.log(continent);

  return (
    <div className="continent">
      <p>Its here</p>
      <div className="home d-flex-row">
        <img
          className="map"
          src=""
          alt="Mapa Mundi"
        />
        <div>
          <h1 className="title">Countries of the World</h1>
          <h2 className="description">
            Click on a continent to see the countries of that continent.
          </h2>
        </div>
      </div>
      {/* {continents.map((continent) => (
        <NavLink
          to={`/${continent.EnglishName}`}
          className="continent"
          key={continent.EnglishName}
        >
          {continent.EnglishName}
        </NavLink>
      ))} */}

    </div>
  );
};

export default Continent;
