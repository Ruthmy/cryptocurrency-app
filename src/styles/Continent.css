.cities__link {
  width: 100%;
  height: 8rem;
  background-color: var(--blue-color);
  background-image: url(../assets/img/circle-right-regular.svg);
  background-position: top 50% right 0.5rem;
  background-size: 2rem;
  background-repeat: no-repeat;
}

.cities__link:nth-child(2n) {
  background-color: var(--blue-color-dark);
}

.cities__content {
  width: 100%;
  height: 8rem;
  justify-content: space-between;
  align-items: center;
  padding: 1rem 4rem 1rem 1rem;
}

.division h2 {
  text-transform: uppercase;
}
