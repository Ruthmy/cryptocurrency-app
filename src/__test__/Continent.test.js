import React from 'react';
import { render, screen } from '@testing-library/react';
import { MemoryRouter, Route, Routes } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import '@testing-library/jest-dom';

import Continent from '../components/Continent';

const mockStore = configureStore([]);
let store;

// Helper function to wrap the component in MemoryRouter for testing NavLink
const renderWithRouter = (component) => render(
  <Provider store={store}>
    <MemoryRouter initialEntries={['/Asia']}>
      <Routes>
        <Route path="/:continent" element={component} />
      </Routes>
    </MemoryRouter>
  </Provider>,
);

describe('Continent component', () => {
  beforeEach(() => {
    store = mockStore({
      weather: {
        weather: [{
          Key: '28143',
          LocalizedName: 'Dhaka',
          EnglishName: 'Dhaka',
          Country: { ID: 'BD', LocalizedName: 'Bangladesh', EnglishName: 'Bangladesh' },
          TimeZone: {
            Code: 'BDT', Name: 'Asia/Dhaka', GmtOffset: 6, IsDaylightSaving: false, NextOffsetChange: null,
          },
          GeoPosition: { Latitude: 23.7098, Longitude: 90.40711, Elevation: { Metric: { Value: 5, Unit: 'm', UnitType: 5 }, Imperial: { Value: 16, Unit: 'ft', UnitType: 0 } } },
          LocalObservationDateTime: '2023-08-05T08:37:00+06:00',
          EpochTime: 1691203020,
          WeatherText: 'Cloudy',
          WeatherIcon: 7,
          HasPrecipitation: false,
          PrecipitationType: null,
          IsDayTime: true,
          Temperature: { Metric: { Value: 28.5, Unit: 'C', UnitType: 17 }, Imperial: { Value: 83, Unit: 'F', UnitType: 18 } },
          MobileLink: 'http://www.accuweather.com/en/bd/dhaka/28143/current-weather/28143?lang=en-us',
          Link: 'http://www.accuweather.com/en/bd/dhaka/28143/current-weather/28143?lang=en-us',
        }],
        status: 'succeeded',
        error: null,
      },
    });
  });

  test('Should have the continent name in the header/home section', () => {
    const { getByText } = renderWithRouter(<Continent />);
    const titleElement = getByText('Asia');
    expect(titleElement).toBeInTheDocument();
  });

  test('Should have the amount of cities in the header/home section', () => {
    renderWithRouter(<Continent />);
    const descriptionElement = screen.getByText('1 cities');
    expect(descriptionElement).toBeInTheDocument();
  });

  test('Should have render the division section', () => {
    renderWithRouter(<Continent />);
    const divisionElement = screen.getByText('LIST OF CITIES IN Asia');
    expect(divisionElement).toBeInTheDocument();
  });

  test('Should have the continent image', () => {
    renderWithRouter(<Continent />);
    const continentImageIsPresent = screen.getByAltText('Asia');
    expect(continentImageIsPresent).toBeInTheDocument();
  });

  test('Should have the city Name and its temperature', () => {
    renderWithRouter(<Continent />);
    const cityIsPresent = screen.getByText('Dhaka');
    const temperatureIsPresent = screen.getByText('28.5 CÂ°');
    expect(cityIsPresent).toBeInTheDocument();
    expect(temperatureIsPresent).toBeInTheDocument();
  });

  test('Should render correctly', () => {
    const { container } = renderWithRouter(<Continent />);
    expect(container).toMatchSnapshot();
  });
});
