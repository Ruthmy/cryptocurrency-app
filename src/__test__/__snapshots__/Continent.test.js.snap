// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Continent component Should render correctly 1`] = `
<div>
  <div>
    <div
      class="home d-flex-row"
    >
      <img
        alt="Asia"
        class="header"
        src="Asia.svg"
      />
      <div
        class="home__info"
      >
        <h1
          class="title"
        >
          Asia
        </h1>
        <p
          class="description"
        >
          1
           cities
        </p>
      </div>
    </div>
    <div
      class="division"
    >
      <h2>
        LIST OF CITIES IN 
        Asia
      </h2>
    </div>
    <div>
      <a
        class="cities__link d-flex-row"
        href="/Asia/Dhaka"
      >
        <div
          class="cities__content d-flex-row"
        >
          <p>
            Dhaka
          </p>
          <p>
            28.5
             
            C
            °
          </p>
        </div>
      </a>
    </div>
  </div>
</div>
`;
